code/modules/virus2/centrifuge.dm

```diff
@@ -42,11 +46,20 @@
 
 	attack_hand(user)
 
+//Also handles luminosity
 /obj/machinery/centrifuge/update_icon()
-	..()
-	if(! (stat & (BROKEN|NOPOWER)) && (isolating || curing))
-		light_color = LIGHT_COLOR_CYAN
-		icon_state = "centrifuge_moving"
+	if(stat & BROKEN)
+		icon_state = "[initial(icon_state)]b"
+		set_light(0, 0, null)
```

Comment:
You don't need to change the `light_power`, nor do you need to change the `light_color`, since it'll only be lit if it's active.